### DISCUSSION TITLE
Dozens of lxqt-config-monitor processes are started – and seem to crash

### DISCUSSION THREAD

**Discussion by l3u** [lxqt/lxqt#2507] [Q&A]
Hi All!

I'm using LxQt on Devuan Daedalus, the version provided there is 1.2.0.

As soon as the energy management turns off the monitor, a lot of "lxqt-config-monitor -l" processes are started. Maybe, the process keeps crashing and is started over and over again. At the time I saw it, the system was up several weeks, and I had > 200 such processes:

    $ ps -ef | grep lxqt-config-monitor
    ...
    serverh+ 19102  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19112  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19122  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19132  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19143  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19153  3339  0 16:30 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19163  3339  0 16:31 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19173  3339  0 16:31 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19183  3339  0 16:31 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 19193  3339  0 16:31 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    ...

Playing around a bit, I saw that at the exact moment the energy management turns off the monitor, a few such processes appear.

I never changed any screen setting. Running "lxqt-config-monitor -l" manually doesn't crash and returns normally.

As soon as the monitor is turned off, the following appears over and over again in ~/.xsession-errors (it was > 300 MB big at the time I noticed this):

    lxqt-session: "Session 'session': display device '/dev/dri/card0'"
    [load.applyBestSettings()] Finished
    qt.qpa.xcb: QXcbConnection: XCB error: 148 (Unknown), sequence: 190, resource id: 0, major code: 140 (Unknown), minor code: 20
    lxqt-session: "Session 'session': display device '/dev/dri/card0'"
    [load.applyBestSettings()] Finished
    lxqt-config-monitor: Applying current settings...
    Output:  "DP-1"
    Output:  "HDMI-1"
    Output:  "DP-2"
    Output:  "HDMI-2"
    lxqt-config-monitor: Settings applied.

What may cause this? And how can I prevent it other than turning off the energy management or removing the executable bit of /usr/bin/lxqt-config-monitor?

Thanks for all help!

**Comment by l3u** [comment]
I investigated this further, and it doesn't matter if the energy management turns the monitor off, or it it's done by xscreensaver. In both cases, I get lxqt-config-monitor zombie processes.

I tried to work around it by moving the lxqt-config-monitor executable and creating a fake script called lxqt-config-monitor that does nothing apart from logging something wanted to start it. Interestingly, now this very script becomes the zombie process. So apparently, lxqt-config-monitor itself isn't even the problem …

I played around with another machine running the same Devuan and LxQt version. Nothing strange happens there. lxqt-config-monitor isn't even started when the monitor is turned off.

So why does this even happen? And what is starting lxqt-config-monitor?!

**Comment by l3u** [comment]
Even if I turn off my monitor manually, I get one `/usr/bin/lxqt-config-monitor -l` zombie process!

What is triggering that command?!

**Reply by tsujan** [reply]
Your problem may have come from Debian Bookworm, on which Devuan Daedalus is based. See https://github.com/lxqt/lxqt/discussions/2489 for a similar problem.

I think a Debian place may be a better place for reporting and discussing it

**Reply by l3u** [reply]
Thanks for the hint. This is a really strange issue. I previously ran LxQt on Gentoo for several years, but on other hardware. Maybe actually some Debian problem …

**Reply by tsujan** [reply]
It may be about an incompatibility related to the old LXQt packages in Bookworm. To rule out this possibility, you might want to upgrade LXQt to a more recent version *if there is a reliable repository for it somewhere* (sorry, I'm not a Debian user).

In old days, when Alf was active, you could rely on LXQt packages that came with Debian. That changed after his disappearence.

**Reply by l3u** [reply]
It's such a shame. In the last almost 20 years, I never experienced something like this using Gentoo. I'm pretty new to that Debian/Devuan stuff. I think it's weird in some places, but I always thought the stable version would be rock-solid. Apparently, this is not the case :-(

**Reply by tsujan** [reply]
Debian itself is a very good and flexible distro — I used its Testing+Unstable versions for 8 years. But an LTS version may have problems that remain with it until the next LTS version; it depends on whether package maintainers are active and other factors.

Anyway, what I said in my previous comment was a wild guess, based on my experience with Debian in old times.
